libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -1298,136 +1442,144 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> im
+                    expectedChoices = [];
                     choice.isExpected = true;
-                    choice.alternative = undefined;
-                    expectedChoices.push(choice.property);
-                    this.addMapping(choice, assignments);
-                    choices = choices.filter((c): boolean => !EntityInfo.overlaps(c, choice.entity));
-                    mapped = true;
-                }
-            }
+                    choice.alternative = null;
+                    if (choice.property) {
+                        expectedChoices.push(choice.property);
+                    } else if (choice.expectedProperties) {
+                        expectedChoices.concat(choice.expectedProperties);
+                    }
 
-            for (const alternative of alternatives) {
-                if (!alternative.operation) {
-                    alternative.operation = this.defaultOperation(alternative, askDefaultOp, defaultOp);
+                    this.addAssignment(choice, assignments);
+                    choices = choices.filter((c) => !EntityInfo.overlaps(c, choice.value));
+                    mapped = true;
                 }
             }
 
             candidate.addAlternatives(alternatives);
-
             if (!mapped) {
-                this.addMapping(candidate, assignments);
+                this.addAssignment(candidate, assignments);
             }
         }
 
-        if (expectedChoices.length > 0) {
+        if (expectedChoices !== null) {
             // When choosing between property assignments, make the assignments be expected.
             actionContext.state.setValue(DialogPath.expectedProperties, expectedChoices);
 
-            // Add back in any non-overlapping choices
-            while (choices.length > 0) {
+            if (expectedChoices.length) {
+                actionContext.state.setValue(DialogPath.expectedProperties, expectedChoices);
+            }
+            // Add back in any non-overlapping choices that have not been resolved.
+            while (choices.length) {
                 const choice = choices[0];
-                const overlaps = choices.filter((alt): boolean => !EntityInfo.overlaps(choice.entity, alt.entity));
+                const overlaps = choices.filter((alt) => EntityInfo.overlaps(choice, alt));
                 choice.addAlternatives(overlaps);
-                this.addMapping(choice, assignments);
-                choices = choices.filter((c): boolean => !EntityInfo.overlaps(c.entity, choice.entity));
+                this.addAssignment(choice, assignments);
+                choices = choices.filter((c): boolean => !EntityInfo.overlaps(c, choice.value));
             }
 
             existing.dequeue(actionContext);
         }
 
-        this.mergeAssignments(assignments, existing);
-        return Object.values(usedEntities);
+        // eslint-disable-next-line prettier/prettier
```

Comment:
Why is this disabled?

Reply:
Not sure. Must've been a bad `ctrl+.` that I missed.